src/openrct2/world/Footpath.cpp

```diff
@@ -1272,148 +1272,210 @@ static bool get_next_direction(int32_t edges, int32_t* direction)
  *              (1 << 5): Unown
  *              (1 << 7): Ignore no entry signs
  */
-static int32_t footpath_is_connected_to_map_edge_recurse(
-    const CoordsXYZ& footpathPos, int32_t direction, int32_t flags, int32_t level, int32_t distanceFromJunction,
+static int32_t footpath_is_connected_to_map_edge_helper(
+    CoordsXYZ footpathPos, int32_t direction, int32_t flags, int32_t level, int32_t distanceFromJunction,
```

Comment:
level, distanceFromJunction, junctionTolerance do not need to be parameters any more and can become local function variables.